controllers/volumereplicationgroup_controller.go

```diff
@@ -309,6 +316,12 @@ func (v *VRGInstance) processVRG() (ctrl.Result, error) {
 		return ctrl.Result{}, nil
 	}
 
+	if err := v.restorePVs(); err != nil {
```

Comment:
This should move to `processAsPrimary`. We only need to restore PVs when we are reconciling as primary and not otherwise (secondary/delete etc.)

Reply:
But the DRPC will never create a VRG as secondary.  Furthermore, if VRG moves from Primary to Secondary, the vrgCondition will not be nil.  So I think it is fine where it is. But I agree, it is better to be sure and specific.

Reply:
It is related to the condition check in the restore, so when that is fixed, we should move this out, or move it out anyway as we get more confident of this approach

Reply:
Right, but we will not apply the PV metadata when the VRG is moved to secondary.  The status condition will be there.  In any case, I agree with both of you that moving it to `processAsPrimary` is the safest.